docker/entrypoint.sh

```diff
@@ -1,22 +1,61 @@
-#!/bin/bash
-
-# Run Django migrations scripts
-python3  /tram/src/tram/manage.py makemigrations tram
-python3  /tram/src/tram/manage.py migrate
+#!/usr/bin/env bash
 
 # Create a super user for the Django server
 #   This should be set by ENV and must include:
 #      - DJANGO_SUPERUSER_USERNAME
 #      - DJANGO_SUPERUSER_PASSWORD
 #      - DJANGO_SUPERUSER_EMAIL
 
-python3 /tram/src/tram/manage.py createsuperuser --noinput
-python3 -c "import nltk; nltk.download('punkt'); nltk.download('wordnet')"
+# Set SKIP_CREATE_SUPERUSER to 1 to skip creating the super user on every run
+
+# check for required environment variables
+SKIP_CREATE_SUPERUSER=${SKIP_CREATE_SUPERUSER:-0}
+
+# only check for required variables if we are not skipping the creation of the superuser
+if [ "${SKIP_CREATE_SUPERUSER}" -eq 0 ]; then
+    if [[ -z "${DJANGO_SUPERUSER_USERNAME:+x}" ]]; then
+        echo "Environment variable DJANGO_SUPERUSER_USERNAME is not set. Skipping superuser creation."
+        SKIP_CREATE_SUPERUSER=1
+    fi
+    if [[ -z "${DJANGO_SUPERUSER_PASSWORD:+x}" ]]; then
+        echo "Environment variable DJANGO_SUPERUSER_PASSWORD is not set. Skipping superuser creation."
+        SKIP_CREATE_SUPERUSER=1
+    fi
+    if [[ -z "${DJANGO_SUPERUSER_EMAIL:+x}" ]]; then
+        echo "Environment variable DJANGO_SUPERUSER_EMAIL is not set. Skipping superuser creation."
+        SKIP_CREATE_SUPERUSER=1
+    fi
+fi
+
+# Generate and Run Django migrations scripts
+python3 /tram/src/tram/manage.py makemigrations tram
+python3 /tram/src/tram/manage.py migrate
+
+# Used provided superuser credentials to create a superuser
+if [[ "${SKIP_CREATE_SUPERUSER}" -eq 0 ]]; then
+
+    PY_CREATE_SU_SCRIPT="
+from django.contrib.auth.models import User;
+
+username = '$DJANGO_SUPERUSER_USERNAME';
+password = '$DJANGO_SUPERUSER_PASSWORD';
+email = '$DJANGO_SUPERUSER_EMAIL';
+
+if not User.objects.filter(username = username).exists():
+    User.objects.create_superuser(username, email, password);
+    print('Superuser created.');
+else:
+    print('Superuser creation skipped, user already exists.');
+"
+    printf "%s" "${PY_CREATE_SU_SCRIPT}" | python3 /tram/src/tram/manage.py shell
```

Comment:
Would this be better done with the django createsuperuser command? Whats the reasoning for doing it this way instead?

Reply:
I think I added this for two reasons: 
 1. Avoid the annoying `CommandError: Error: That username is already taken.` message, 
 2. Provide a useful message to user if an environment variable was not provided. By default just provides an ambiguous error.

It could be done in Python, but would require creating a `createsuperuser` wrapper to handle the conditions, so I just added it here to be faster.